### PR TITLE
LRDOCS-5109 liferay frontend taglibs

### PR DESCRIPTION
**Pull request by mwilliams2014**
@jbalsas Here are the liferay-frontend taglib tutorials I have so far. Let me know what you think. I still haven't documented the tags below. Please feel free to point to code examples for these tags, or provide some and I'll add tutorials for those as well:
- defineObjects
- diff
- diff-version-comparator
- email-notifcation-settings
- translation-manager
And I didn't document the `liferay-frontend:image-card` tag because I couldn't find any example use cases in Portal. Is the clay:image-card tag preferred over this because several of those are in Portal's code?

### CODE REVIEW

File: develop/tutorials/articles/210-front-end-taglibs/00-front-end-taglibs-intro.markdown [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
 [Clay components](https://clayui.com/docs/components/alerts.html), 
-such as alerts, buttons, drop-down menus, form elements, and more to your apps.
-The chart taglib visualizes data: you can create bar charts, line charts,
-scatter charts, spline charts, and much more.
+  such as alerts, buttons, drop-down menus, form elements, and more to your apps. 
+
+- Frontend: Add menus, visualize information, manage search container results, 
+  and more. 
+
+- Liferay UI: lets you create common UI components such as icons, tabs, and
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/00-front-end-taglibs-intro.markdown]: This classification looks odd, but I don't have anything better to offer... 🤷‍♂️

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/00-using-liferay-frontend-taglibs-intro.markdown [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+The Liferay Frontend tag library provides a set of tags for creating common 
+frontend UI components in your app.  
+
+To use the Frontend taglib, add the following declaration to your JSP:
+
+    <%@ taglib prefix="liferay-frontend" uri="http://liferay.com/tld/frontend" %>
+
+The following Frontend UI components are covered in this section of tutorials:
+
+- [Add Menu](/develop/tutorials/-/knowledge_base/7-1/liferay-frontend-add-menu)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/00-using-liferay-frontend-taglibs-intro.markdown]: We should mark this as deprecated for `7.1` and direct to the `clay:management-toolbar` one

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/00-using-liferay-frontend-taglibs-intro.markdown [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- [Cards](/develop/tutorials/-/knowledge_base/7-1/liferay-frontend-cards)
+- [Info Bar](/develop/tutorials/-/knowledge_base/7-1/liferay-frontend-info-bar)
+- [Management Bar](/develop/tutorials/-/knowledge_base/7-1/liferay-frontend-management-bar)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/00-using-liferay-frontend-taglibs-intro.markdown]: We should mark this as deprecated for `7.1` and direct to the `clay:management-toolbar` one

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/01-liferay-frontend-add-menu.markdown [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# Liferay Frontend Add Menu [](id=liferay-frontend-add-menu)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/01-liferay-frontend-add-menu.markdown]: We should mark this as deprecated for `7.1` and direct to the `clay:management-toolbar` one. We have a migration pattern in https://ui-migration.liferay.com/docs/lexicon/add_menu.html that might be useful

Other than that, this one looks great! 👍

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/02-liferay-frontend-cards.markdown [hunk 5]
```diff
@@ -0,0 +1,116 @@
+# Liferay Frontend Cards [](id=liferay-frontend-cards)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/02-liferay-frontend-cards.markdown]: Nice! 👍

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/03-liferay-frontend-info-bar.markdown [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    </liferay-frontend:info-bar>
+
+The `<liferay-frontend:info-bar-sidenav-toggler-button>` tag uses 
+[Clay Icons](/develop/tutorials/-/knowledge_base/7-1/clay-icons) 
+for the `icon` attribute. 
+
+sidebar panel:
+
+    <div class="closed container-fluid-1280 sidenav-container sidenav-right" id="<portlet:namespace />infoPanelId">
+    <liferay-frontend:sidebar-panel>
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/03-liferay-frontend-info-bar.markdown]: Fix code indentation here

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/00-liferay-frontend-management-bar-intro.markdown [hunk 7]
```diff
@@ -0,0 +1,94 @@
+# Liferay Frontend Management Bar [](id=liferay-frontend-management-bar)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/00-liferay-frontend-management-bar-intro.markdown]: We should mark this as deprecated for `7.1` and direct to the `clay:management-toolbar` one. We also have several migration patterns documented in https://ui-migration.liferay.com/docs/lexicon/management_toolbar/selectable.html that might prove useful.

Other than that, it looks great!

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/01-creating-display-options-for-your-results.markdown [hunk 8]
```diff
@@ -0,0 +1,187 @@
+# Creating Display Options for Search Results [](id=creating-display-options-for-search-results)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/01-creating-display-options-for-your-results.markdown]: This feels a bit out of scope here... I like the style and it is very valuable information, but something tells me this belongs somewhere else, not inside the "Taglibs" category... 🤔

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/02-filtering-search-results.markdown [hunk 9]
```diff
@@ -0,0 +1,74 @@
+# Filtering Search Results with the Management Bar [](id=filtering-search-results-with-the-management-bar)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/02-filtering-search-results.markdown]: Similar comment as above... this is not something specifically related to "using a taglib", but rather how to implement a specific behaviour. Feels a bit out of scope for this section.

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/00-sorting-search-results-intro.markdown [hunk 10]
```diff
@@ -0,0 +1,88 @@
+# Sorting Search Results [](id=sorting-search-results-with-the-management-bar)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/00-sorting-search-results-intro.markdown]: Same thing here... and the subsequent nesting that is happening after this should reinforce that, I think 🤔

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/01-creating-the-comparator-class.markdown [hunk 11]
```diff
@@ -0,0 +1,128 @@
+# Creating the Comparator Class [](id=creating-the-comparator-class)
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/01-creating-the-comparator-class.markdown]: This has definitely nothing to do with tags.

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/04-including-actions-in-the-management-bar.markdown [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    management bar. A code snippet from the Site admin portlet is shown below: 
+
+        <liferay-frontend:management-bar-action-buttons>
+        	<liferay-frontend:management-bar-sidenav-toggler-button
+        		icon="info-circle"
+        		label="info"
+        	/>
+
+        	<liferay-frontend:management-bar-button
+        		href="javascript:;"
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/04-including-actions-in-the-management-bar.markdown]: We should show here either:
- A button with a proper `href`
- The JS code that makes that button do something

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/05-disabling-the-management-bar.markdown [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+You can also disable individual components by adding the `disabled` attribute to 
+the corresponding tag. The example below disables the display buttons when the 
+search container displays 0 results, since changing the display style has no 
+effect when there aren't any results to view:
+
+    <liferay-frontend:management-bar-display-buttons
+            disabled="<%= total == 0 %>"
+            displayViews='<%= new String[] {"descriptive", "icon", "list"} %>'
+            portletURL="<%= changeDisplayStyleURL %>"
+            selectedDisplayStyle="<%= displayStyle( %>"
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/05-disabling-the-management-bar.markdown]: Missing code?

File: develop/tutorials/articles/210-front-end-taglibs/00-front-end-taglibs-intro.markdown (hunk 1 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/00-front-end-taglibs-intro.markdown]: I'll try to improve the description a bit

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/01-liferay-frontend-add-menu.markdown (hunk 4 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/01-liferay-frontend-add-menu.markdown]: Ah, ok. Will do. Hopefully the changes are not too drastic so I don't have to rewrite these docs from scratch.

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/01-creating-display-options-for-your-results.markdown (hunk 8 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/01-creating-display-options-for-your-results.markdown]: It directly relates to the management bar, at least the frontend management bar, but that may not be the case for the clay:management-bar...So, I'm not sure where else I would put the information. You need to know it to be able to configure the displaystyles attribute for the taglib. I'll try to see if perhaps it can be reorganized after I make the changes to clay:management-bar.

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/02-filtering-search-results.markdown (hunk 9 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/02-filtering-search-results.markdown]: Same comment as above, but since these two are perhaps more related to the lexicon patterns to follow, it can be moved to the (yet to be ported from 7.0) tutorial section on Applying Clay Styles to Your Portlet. That way we can just keep this section purely taglib and attributes.

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/01-creating-the-comparator-class.markdown (hunk 11 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/01-creating-the-comparator-class.markdown]: Again, this is required to create the sorting filter which is used in the management bar, so we definitely want to share this information, but I agree it is a bit heavy for this section. This should be placed in the Applying Clay Styles to Your Apps as well. I moved some of these tutorials over from the previous Applying Lexicon Styles to Your Apps 7.0 tutorials because they were so closely related to the management bar. I think those will just have to stay in their own section though and leave this section to purely configuring the taglibs.

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/05-disabling-the-management-bar.markdown (hunk 13 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/05-disabling-the-management-bar.markdown]: haha, yeah looks like some may have not made it. I'll fix that. Thanks!

File: develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/01-creating-the-comparator-class.markdown (hunk 11 above)
Comment (jbalsas) [reply develop/tutorials/articles/210-front-end-taglibs/02-liferay-frontend-taglibs/04-liferay-frontend-management-bar/03-sorting-search-results/01-creating-the-comparator-class.markdown]: Could we please keep these in the backburner for now? Let's just keep pushing simple and direct taglib ones and figure out where everything fits later.

**Comment by jbalsas** [comment]
Closing here as you said you'll resend when ready ;)

**Comment by mwilliams2014** [comment]
Sent https://github.com/jbalsas/liferay-docs/pull/24 with updates
